### PR TITLE
Replace KAPT with KSP

### PR DESCRIPTION
**Pull request by zsmb13**
### 🎯 Goal

Migrate from using KAPT to KSP, as KSP is stable now.

Blocker:

- Weird runtime crash in Room converters

        com.squareup.moshi.JsonDataException: Expected BEGIN_OBJECT but was STRING at path $[0]
        at com.squareup.moshi.JsonUtf8Reader.beginObject(JsonUtf8Reader.java:172)
        at com.squareup.moshi.MapJsonAdapter.fromJson(MapJsonAdapter.java:69)
        at com.squareup.moshi.MapJsonAdapter.fromJson(MapJsonAdapter.java:30)
        at com.squareup.moshi.internal.NullSafeJsonAdapter.fromJson(NullSafeJsonAdapter.java:41)
        at com.squareup.moshi.CollectionJsonAdapter.fromJson(CollectionJsonAdapter.java:81)
        at com.squareup.moshi.CollectionJsonAdapter$2.fromJson(CollectionJsonAdapter.java:55)
        at com.squareup.moshi.internal.NullSafeJsonAdapter.fromJson(NullSafeJsonAdapter.java:41)
        at com.squareup.moshi.JsonAdapter.fromJson(JsonAdapter.java:70)
        at io.getstream.chat.android.offline.repository.database.converter.QuerySortConverter.stringToObject(QuerySortConverter.kt:18)

### 🧪 Testing

Run tests, smoke test the app.

### ☑️ Checklist

- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] PR targets the `develop` branch
- [ ] ~Changelog is updated with client-facing changes~
- [ ] ~New code is covered by unit tests~
- [ ] ~Comparison screenshots added for visual changes~
- [ ] ~Affected documentation updated (docusaurus, tutorial, CMS (task created))~
- [ ] Reviewers added

### 🎉 GIF

_Please provide a suitable gif that describes your work on this pull request_

### CODE REVIEW

**Comment by skydoves** [comment]
Seems Room [2.4.0-rc01](https://android-review.googlesource.com/c/platform/frameworks/support/+/1896841/1/buildSrc/public/src/main/kotlin/androidx/build/LibraryVersions.kt#107) will be released soon. 👀

File: stream-chat-android-offline/build.gradle [hunk 1]
```diff
@@ -45,10 +45,8 @@ android {
     }
 }
 
-kapt {
-    arguments {
-        arg("room.incremental", "true")
-    }
+ksp {
+    arg("room.useNullAwareTypeAnalysis", "false")
```
Comment (skydoves) [inline stream-chat-android-offline/build.gradle:49]: Room [2.4.0-beta02](https://developer.android.com/jetpack/androidx/releases/room#2.4.0-beta02) introduced a new `TypeConverter analyzer` that takes nullability information in types into account. We should turn this option off for avoiding the type error.

Comment (adasiewiczr) [reply stream-chat-android-offline/build.gradle:49]: I'll add a task to remove this as flag will be removed in the future

**Review by adasiewiczr — APPROVED** [review]
### ☑️Reviewer Checklist
- [x] UI Components sample runs & works
- [x] Compose sample runs & works
- [ ] UI Changes correct (before & after images)
- [ ] Bugs validated (bugfixes)
- [ ] New feature tested and works
- [ ] Release notes and docs clearly describe changes
- [ ] All code we touched has new or updated KDocs

File: buildSrc/src/main/kotlin/io/getstream/chat/android/Dependencies.kt [hunk 2]
```diff
@@ -178,8 +180,8 @@ object Dependencies {
     const val mockito = "org.mockito:mockito-core:${Versions.MOCKITO}"
     const val mockitoKotlin = "com.nhaarman.mockitokotlin2:mockito-kotlin:${Versions.MOCKITO_KOTLIN}"
     const val moshi = "com.squareup.moshi:moshi:${Versions.MOSHI}"
-    const val moshiCodegen = "com.squareup.moshi:moshi-kotlin-codegen:${Versions.MOSHI}"
     const val moshiKotlin = "com.squareup.moshi:moshi-kotlin:${Versions.MOSHI}"
+    const val moshiCodegen = "com.squareup.moshi:moshi-kotlin-codegen:${Versions.MOSHI}"
```
Comment (JcMinarro) [inline buildSrc/src/main/kotlin/io/getstream/chat/android/Dependencies.kt:184]: Could we keep alphabetical order here?

Comment (skydoves) [reply buildSrc/src/main/kotlin/io/getstream/chat/android/Dependencies.kt]: Done: 3c25121

**Review by JcMinarro — APPROVED** [review]
LGTM

**Comment by filbabic** [comment]
### ☑️Reviewer Checklist
- [x] UI Components sample runs & works
- [x] Compose sample runs & works
- [ ] UI Changes correct (before & after images)
- [ ] Bugs validated (bugfixes)
- [ ] New feature tested and works
- [ ] Release notes and docs clearly describe changes
- [ ] All code we touched has new or updated KDocs

Looks good! :]

**Pull request merged**